supervisor/api/__init__.py

```diff
@@ -443,13 +510,32 @@ def _register_addons(self) -> None:
                 ),
                 web.get("/addons/{addon}/options/config", api_addons.options_config),
                 web.post("/addons/{addon}/rebuild", api_addons.rebuild),
-                web.get("/addons/{addon}/logs", api_addons.logs),
                 web.post("/addons/{addon}/stdin", api_addons.stdin),
                 web.post("/addons/{addon}/security", api_addons.security),
                 web.get("/addons/{addon}/stats", api_addons.stats),
             ]
         )
 
+        async def get_addon_logs(request, *args, **kwargs):
+            addon: Addon = api_addons.get_addon_for_request(request)
+            kwargs["identifier"] = f"addon_{addon.slug}"
+            return await self._api_host.advanced_logs(request, *args, **kwargs)
```

Comment:
> Or move all the logic in self._api_host.advanced_logs into another method without the annotation and have both places call that and then add the annotation to this method.

This sounds like the best option to me - splitting the `advanced_logs` into a method fetching the logs and and API endpoint handler. However, it creates an inconsistency between the add-on logs endpoints and the other ones, because with `api_process_raw(CONTENT_TYPE_BINARY)` you always get a response with this mimetype, instead of the `text/plain` that is returned normally (and it's also a bit weird you get a different type than what you specified in the `Accept` header IMO). Moreover, "plain" usage of `advanced_logs` annotated with `api_process` returns the content type specified within the function, but errors are returned as `application/json` (this was always the case, not introduced by my PR).

And one more thing - while `api_process_raw` handles the errors sensibly, it does not return bad request - it always returns response with HTTP 200. Also this was always there, and I'm not sure if it should be changed within the scope of this PR, or even changed at all, because I might not see the big picture there.

Reply:
> However, it creates an inconsistency between the add-on logs endpoints and the other ones, because with `api_process_raw(CONTENT_TYPE_BINARY)` you always get a response with this mimetype, instead of the `text/plain` that is returned normally (and it's also a bit weird you get a different type than what you specified in the Accept header IMO). 

Oh that's fair. I'm not sure we currently have a way in this code to react to the `Accept` header but that does make sense. I was mostly concerned about the consistent error handling to be honest.

Perhaps in that case it would be best to make the `web.Response` object in the method and return that? And then make a new annotation that exclusively handles errors. It assumes any non-exception return is already a `web.Response` object and just returns it as is.

> Moreover, "plain" usage of advanced_logs annotated with api_process returns the content type specified within the function, but errors are returned as application/json (this was always the case, not introduced by my PR).

Oh I didn't realize that. I thought the code in there seemed to attempt to encode its return to binary so it was always that, I guess I didn't look at it closely enough. I'm open to suggestions here, just want to make sure we aren't bubbling up exceptions past our code and instead returning error responses.

Reply:
Since I didn't want to break things and `api_proces_raw` doesn't behave in the way I'd like (e.g. it always returns HTTP 200 and doesn't handle all `HassioError` types), I created a new decorator for API functions that's something between `api_process` and `api_process_raw`. I also used it for other endpoints to address the consistency problems I mention above.